branchLayoutImpl/src/main/java/com/virtuslab/branchlayout/impl/manager/BranchLayoutManager.java

```diff
@@ -0,0 +1,61 @@
+package com.virtuslab.branchlayout.impl.manager;
+
+import java.nio.file.Path;
+
+import io.vavr.collection.List;
+import io.vavr.control.Try;
+import lombok.Getter;
+
+import com.virtuslab.branchlayout.api.manager.IBranchLayoutManager;
+import com.virtuslab.branchlayout.api.manager.IBranchLayoutReader;
+import com.virtuslab.branchlayout.api.manager.IBranchLayoutWriter;
+import com.virtuslab.logger.IPrefixedLambdaLogger;
+import com.virtuslab.logger.PrefixedLambdaLoggerFactory;
+
+public class BranchLayoutManager implements IBranchLayoutManager {
+  private static final IPrefixedLambdaLogger LOG = PrefixedLambdaLoggerFactory.getLogger("branchLayout");
+
+  @Getter
+  private final IBranchLayoutReader reader;
+  @Getter
+  private final IBranchLayoutWriter writer;
+
+  public BranchLayoutManager(Path path, boolean isBranchLayoutPresent) {
+    IndentSpec indentSpec = isBranchLayoutPresent ? deriveIndentSpec(path) : IndentSpec.DEFAULT_SPEC;
+    this.reader = new BranchLayoutFileReader(path, indentSpec.getIndentCharacter(), indentSpec.getIndentWidth());
```

Comment:
Maybe make the Reader/Writer c'tor accept the entire `IIndentSpec` object then insted?